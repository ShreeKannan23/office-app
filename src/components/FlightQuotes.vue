<template>
  <div class="col-auto quote-card px-3 pt-3 pb-0 mr-3">
    <div class="row mx-auto quote-type mb-2">
      <p class="m-0">{{ quote.type }}</p>
    </div>
    <div class="row mx-auto py-1">
      <p class="col-5 pl-0 m-0">Operator :</p>
      <p class="col pl-0 m-0">
        {{ quote.carrierName }}
      </p>
    </div>
    <div class="row mx-auto py-1">
      <p class="col-5 pl-0 m-0">Price :</p>
      <p class="col pl-0 m-0">{{ quote.price }}&euro;</p>
    </div>
    <div class="row mx-auto mt-2">
      <p>{{ quote.originCityName }}</p>
      &nbsp;-------<img
        src="../assets/icons/icon_flight_lookup.svg"
        width="24"
        height="24"
      />-------&nbsp;
      <p>{{ quote.destinationCityName }}</p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'FlightQuotes',
  props: {
    quote: {
      type: Object,
      required: true
    }
  }
}
</script>

<style scoped>
.quote-type {
  font-size: 21px;
  font-weight: 500;
  line-height: 21px;
}
</style>
