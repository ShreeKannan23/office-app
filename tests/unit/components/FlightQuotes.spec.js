import { shallowMount } from '@vue/test-utils'
import FlightQuotes from '@/components/FlightQuotes.vue'

describe('FlightQuotes.vue', () => {
  let component = null

  function componentInstance (props) {
    return shallowMount(FlightQuotes, {
      props
    })
  }

  beforeEach(() => {
    component = componentInstance({
      quote: {
        type: 'Cheapest',
        carrierName: 'KLM',
        originCityName: 'Amsterdam',
        destinationCityName: 'Madrid'
      }
    })
  })

  afterAll(() => {
    component.destroy()
  })
  it('check default values', () => {
    expect(component.vm.quote).toMatchObject({
      type: 'Cheapest',
      carrierName: 'KLM',
      originCityName: 'Amsterdam',
      destinationCityName: 'Madrid'
    })
    expect(component).toMatchSnapshot()
  })
})
