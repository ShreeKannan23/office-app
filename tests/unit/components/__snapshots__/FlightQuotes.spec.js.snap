// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightQuotes.vue check default values 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="col-auto quote-card px-3 pt-3 pb-0 mr-3"
      >
        <div
          class="row mx-auto quote-type mb-2"
        >
          <p
            class="m-0"
          >
            Cheapest
          </p>
        </div>
        <div
          class="row mx-auto py-1"
        >
          <p
            class="col-5 pl-0 m-0"
          >
            Operator :
          </p>
          <p
            class="col pl-0 m-0"
          >
            KLM
          </p>
        </div>
        <div
          class="row mx-auto py-1"
        >
          <p
            class="col-5 pl-0 m-0"
          >
            Price :
          </p>
          <p
            class="col pl-0 m-0"
          >
            €
          </p>
        </div>
        <div
          class="row mx-auto mt-2"
        >
          <p>
            Amsterdam
          </p>
            -------
          <img
            height="24"
            width="24"
          />
          -------  
          <p>
            Madrid
          </p>
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": Object {},
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": Object {},
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.3",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "quote": Object {
      "carrierName": "KLM",
      "destinationCityName": "Madrid",
      "originCityName": "Amsterdam",
      "type": "Cheapest",
    },
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
